src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,381 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string>
+
+#include "application_manager/commands/mobile/show_constant_tbt_request.h"
+#include "application_manager/commands/command_request_test.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/message_helper.h"
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace show_constant_tbt_request {
+
+namespace am = application_manager;
+using am::commands::ShowConstantTBTRequest;
+using ::utils::SharedPtr;
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+using policy_test::MockPolicyHandlerInterface;
+
+namespace {
+const uint32_t kAppId = 1u;
+const uint32_t kConnectionKey = 2u;
+const std::string kNavigationText1 = "Some text1";
+const std::string kNavigationText2 = "Some text2";
+const std::string kEta = "Some time";
+const std::string kTotalDistance = "Some distance";
+const std::string kTimeToDestination = "Some time";
+const std::string kTutnIconValue = "0x1A3F";
+const std::string kNextTutnIconValue = "0x1A40";
+}  // namespace
+
+class ShowConstantTBTRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateFullParamsUISO() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    msg_params[am::strings::turn_icon][am::strings::value] = kTutnIconValue;
+    msg_params[am::strings::next_turn_icon][am::strings::value] =
+        kNextTutnIconValue;
+    msg_params[am::strings::navigation_text_1] = kNavigationText1;
+    msg_params[am::strings::navigation_text_2] = kNavigationText2;
+    msg_params[am::strings::eta] = kEta;
+    msg_params[am::strings::total_distance] = kTotalDistance;
+    msg_params[am::strings::time_to_destination] = kTimeToDestination;
+    msg_params[am::strings::soft_buttons] = "sds";
+
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  MessageSharedPtr CreateExpectedMessage() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    int32_t index = 0;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
```

Comment:
@BSolonenko is `static_cast` required here to init smart object?

Reply:
@AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/1977/commits/2127b6028b54933a3f410a26f186212ea4845467